src/cr/cube/cube.py

```diff
@@ -543,12 +593,13 @@ def _slice_idxs(self):
 
     @lazyproperty
     def _valid_idxs(self):
+        """Tuple of int64 ndarrays of the valid elements idx for each dimension."""
         valid_idxs = np.ix_(
             *tuple(d.valid_elements.element_idxs for d in self._all_dimensions)
         )
 
         def _reshape_idxs(valid_indices):
-            if self._measure.requires_array_transposition:
+            if self._all_dimensions.require_transposition:
```

Comment:
Since this is a temporary hack while we wait for `dim_order`, I think it's a mistake to pollute `AllDimensions` when we can easily contain the hackery to this module.

Just create `Cube.requires_array_transposition` (even if it is redundant with `_BaseMeasure.requires_array_transposition`) and mark it as a hack with a comment and when it is expected to go away.